dvc/remote/base.py

```diff
@@ -233,6 +241,9 @@ def _collect_dir(self, path_info):
         return sorted(result, key=itemgetter(self.PARAM_RELPATH))
 
     def get_dir_checksum(self, path_info):
+        if not self.cache:
```

Comment:
Would it make sense to unify it somehow with https://github.com/iterative/dvc/blob/master/dvc/output/base.py#L102 ?

Reply:
@efiop Do you mean, using same error in both places, or using `self.REMOTE.scheme` to point user in right direction?

Reply:
@pared Well, maybe both? :)